### PR TITLE
Enable NVTX profiling in MCore

### PR DESCRIPTION
**Pull request by minitu**
> [!IMPORTANT]  
> The `Update branch` button must only be pressed in very rare occassions.
> An outdated branch is never blocking the merge of a PR.
> Please reach out to the automation team before pressing that button.

# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog

- Add specific line by line info of high level changes in this PR.

# Usage

- You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"

**Pre checks**:

- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:

- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed.
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information

- Related to # (issue)

### CODE REVIEW

File: nemo/lightning/pytorch/callbacks/nsys.py [hunk 1]
```diff
@@ -16,6 +16,7 @@
 
 import torch
 from lightning.pytorch.callbacks.callback import Callback
+from megatron.core.utils import configure_nvtx_profiling
```
Comment (guyueh1) [inline nemo/lightning/pytorch/callbacks/nsys.py]: can you properly handle the import error here, maybe add a if condition on megatron.core version and fall back to None?

File: nemo/lightning/pytorch/callbacks/nsys.py [hunk 2]
```diff
@@ -79,6 +80,8 @@ def __init__(
 
         app_state = AppState()
         app_state._nvtx_ranges = nvtx_ranges
+        if nvtx_ranges:
+            configure_nvtx_profiling(True)  # Enable NVTX profiling in MCore
```
Comment (guyueh1) [inline nemo/lightning/pytorch/callbacks/nsys.py:90]: can you add another flag like "configure_mcore_nvtx_ranges", instead of reusing this one (i think it is meant for inserting some existing nvtx markers in nemo).

Comment (minitu) [reply nemo/lightning/pytorch/callbacks/nsys.py:90]: The `nvtx_ranges` flag here is actually something I added, which enables additional NVTX ranges for performance breakdown. So I think it makes sense to use it to enable the MCore side of adding NVTX ranges as well

Comment (guyueh1) [reply nemo/lightning/pytorch/callbacks/nsys.py:90]: Ok i agree

**Review by guyueh1 — APPROVED** [review]
